Fixed bugs in navigator_middleware.dart

uditrajmr3 opened the pull request:
Fixed: Null cannot be a subtype of Route<dynamic> in type cast

Fixed: A GlobalKey<NavigationState> was used multiple times inside one widget's child list

uditrajmr3 committed:
Fixed bugs in navigator_middleware.dart

Fixed: Null cannot be a subtype of Route<dynamic> in type cast

Fixed: A GlobalKey<NavigationState> was used multiple times inside one widget's child list

payam-zahedi commented:
Hey @uditrajmr3 

Thanks for your contribution

one question, Did you test this on your machine?

payam-zahedi commented on lib/service/navigator_middleware.dart:
```diff
@@ -13,67 +12,64 @@ class NavigatorMiddleware<R extends Route<dynamic>> extends RouteObserver<R> {
     this.onRemove,
   }) : _stack = [];
-
+  final GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
```
Why do we need this navigator key?

uditrajmr3 replied:
It's Optional for setting up with Navigator 2.0 routing. As when using middleware, if one would want to navigate, they can use navigator key. It's also useful for Nested Navigation. 
So, I should have quoted this in Comments.

uditrajmr3 replied:
@payam-zahedi I have updated the code to the best of my knowledge. please care to look into it and update me for any other required changes. Thanks.

uditrajmr3 commented:
> Hey @uditrajmr3 
> 
> Thanks for your contribution
> 
> one question, Did you test this on your machine?

Indeed. MacOS 16 and Windows 11 patch 1.2 Both.

payam-zahedi commented:
Ok thanks for your response

so do you want to update your code? Or can I merge this PR?

uditrajmr3 commented:
> Ok thanks for your response
> 
> so do you want to update your code? Or can I merge this PR?

I will update it soon

uditrajmr3 committed:
Update navigator_middleware.dart

Updated: Explaining Use of NavigatorKey

Added: Comments to explain use of navigator key

payam-zahedi commented:
Well done
Thanks for your contribution again
